contracts/AelinUpFrontDeal.sol

```diff
@@ -0,0 +1,299 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import "./AelinERC20.sol";
+import {AelinDeal} from "./AelinDeal.sol";
+import {AelinPool} from "./AelinPool.sol";
+import {IAelinPool} from "./interfaces/IAelinPool.sol";
+import {IAelinUpFrontDeal} from "./interfaces/IAelinUpFrontDeal.sol";
+import "./libraries/NftCheck.sol";
+import "./interfaces/ICryptoPunks.sol";
+
+contract AelinUpFrontDeal is AelinERC20, IAelinUpFrontDeal {
+    address constant CRYPTO_PUNKS = address(0xb47e3cd837dDF8e4c57F05d70Ab865de6e193BBB);
+
+    UpFrontPool public poolData;
+    UpFrontDeal public dealData;
+
+    mapping(address => uint256) public allowList;
+    mapping(address => IAelinPool.NftCollectionRules) public nftCollectionDetails;
+    mapping(address => mapping(address => bool)) public nftWalletUsedForPurchase;
+    mapping(address => mapping(uint256 => bool)) public nftId;
+    mapping(address => uint256) public amountAccepted;
+
+    uint256 public totalAmountAccepted;
+    uint256 public currentDealTokenTotal;
+    uint256 private poolStart;
+
+    bool private hasAllowList;
+    bool private hasNftList;
+    bool private dealDepositComplete;
+    bool public poolDepositComplete;
+
+    function initialize(
+        UpFrontPool calldata _poolData,
```

Comment:
can we combine all these arguments into a single struct since the pool and deal are essentially the same thing here so no reason to break them apart. same with the last 2 arguments 
```uint256 _underlyingDealTokenAmount,
address _sponsor```